src/contract/assembled_transaction.ts

```diff
@@ -609,33 +608,36 @@ export class AssembledTransaction<T> {
     force = false,
     signTransaction = this.options.signTransaction,
   }: {
-    /**
-     * If `true`, sign and send the transaction even if it is a read call
-     */
+        /**
+         * If `true`, sign and send the transaction even if it is a read call
+         */
     force?: boolean;
-    /**
-     * You must provide this here if you did not provide one before
-     */
+        /**
+         * You must provide this here if you did not provide one before
+         */
     signTransaction?: ClientOptions["signTransaction"];
   } = {}): Promise<void> => {
     if (!this.built) {
-      throw new Error("Transaction has not yet been simulated");
+      throw new Error("Transaction has not yet been simulated.");
     }
-
+  
     if (!force && this.isReadCall) {
       throw new AssembledTransaction.Errors.NoSignatureNeeded(
-        "This is a read call. It requires no signature or sending. " +
-        "Use `force: true` to sign and send anyway."
+        "This is a read call. It requires no signature or sending. Use `force: true` to sign and send anyway."
       );
     }
-
+  
     if (!signTransaction) {
       throw new AssembledTransaction.Errors.NoSigner(
-        "You must provide a signTransaction function, either when calling " +
-        "`signAndSend` or when initializing your Client"
+        "You must provide a `signTransaction` function, either when calling `signAndSend` or when initializing your Client."
+      );
+    }
+    // Ensure publicKey is provided for non-read calls
+    if (!this.options.publicKey) {
+      throw new Error(
+        "Public key not provided. Have you forgotten to set the `publicKey` in AssembledTransactionOptions?"
```

Comment:
It looks like this is the only needed change here. Everything except this block seems either like anti-linting or outright error. Maybe there's a test of this specific logic that should stay; I again didn't get that far.

Can you please force-push a single commit to this branch with only these specific changes? We do not want to merge anything else.

I see you pushed right to your `master` branch so it might be a little more complicated and error-prone, but here's a way you can do it:

- `log --pretty=oneline -n 10 --graph` – this should show you the commit, eight commits ago, where you branched off what was then `master` on the upstream remote (not sure how your local remotes are named). Copy that commit hash.
- `git reset --hard [pasted commit hash]` – this will reset all local changes to your branch
- `git pull upstream master` – may as well update to the tip of the upstream branch to avoid as many conflicts as possible. Again, not sure what your local remotes are called. You can use `git remote -v` to see all remotes and make sure you have the name of the `stellar/js-stellar-sdk` one, and slot it in where I put the `upstream`.
- Now you can copy-paste specific changes from this PR back into your local changes and commit them with a [sensible commit message](https://cbea.ms/git-commit/).
- `git push --force origin master` – one more time, not sure what your remotes are called. Make sure to use whatever name you gave your remote, rather than `origin`.